src/adaptix/_internal/model_tools/introspection/typed_dict.py

```diff
@@ -36,12 +38,35 @@ def _get_td_hints(tp):
     return elements
 
 
-if HAS_PY_39:
-    def _make_requirement_determinant(tp):
+if HAS_PY_311:
+    def _make_requirement_determinant(required_fields: set):
+        return lambda name: name in required_fields
+
+    def _extract_item_type(tp) -> BaseNormType:
+        if tp.origin is typing.Annotated:
+            return tp.args[0]
+        return tp
+
+    def _fetch_required_keys(
+        fields: Sequence[Tuple[str, BaseNormType]],
+        frozen_required_keys: frozenset[str],
```

Comment:
There is no guarantee that ``__required_keys__`` must be frozenset instance, so it would be better to use ``AbstactSet`` type hint